File: hw-1B/C++/D-school.cpp
#include <iostream>

using namespace std;

int main()
{
    /*  Input: n = number of students, coord = coordinates of students' houses.
        Find the coordinate of the school to minimize the sum of distances 
        from houses to the school of all students. */

    int n;
    cin >> n;
    int coord[n];
    for (int i = 0; i < n; i++){
		cin >> coord[i];
	}
    int res =  n / 2;
    cout << coord[res] << endl;
    return 0;
}

File: hw-1B/Python/D-school.py
def school_location(arr):
    
    m = len(arr) // 2
    return m


if __name__ == '__main__':

    """ Input: n = number of students, coord = coordinates of students' houses.
        Find the coordinate of the school to minimize the sum of distances 
        from houses to the school of all students. """

    n = int(input())
    coord = [int(x) for x in input().split()]
    res = school_location(coord)
    print(coord[res])